### DISCUSSION TITLE
Allow for predefined time duration in "Override", instead of open input

### DISCUSSION THREAD

**Discussion by knuurr** [proginosko/LeechBlockNG#520] [Ideas]
Current state: when hitting Override, an window appears which asks for how long do you want current block to be overridden

My suggestion: allow to predefine a time range for which Override can be enabled. This could be done in few ways:

- user defines a time range for which Override an be activated. Input on "Override" page is reduced from manual typing to hitting a button. 
- There could be a single button, buttons (if more than one time range is allowed to be defined) or using select dropdown menu.
- An open input box appears along with buttons(s) for predefined time ranges.

Additionally:
- if time ranges are enabled and defined, maybe let's add option to disable open input option completely
- to avoid perpetual coming back and hitting "5 minutes" or other option, a slights delay (5-10s) might be added before time range can be chosen

here's an example: user defined their own time ranges for override, 5/10/15 minutes. They also decided to leave "custom time" enabled (ideally, they can decide to toggle it on/off). Only after they pick time for Override, a counter starts on "Activate Override" after which a buton will switch state from deactivated to clickable (time might be configurable, with 0 as value to disable it). This is to penalise compulsive clicking for another 5/10/15 minutes every time Override finishes.

here is a small wireframe of how might it look. It is up to discussion of course.

![obraz](https://github.com/user-attachments/assets/a602ba61-277b-4c5d-8ea5-157e7cc20cec)

I didn't find a way to do it with current version. And I think it mirrors well how it's done for example for apps on Android.

**Comment by proginosko** [comment]
You can prespecify a duration under Options > General > Temporary Override. If you leave it blank, then you get the prompt to enter a duration at the time the override is activated.

**Reply by knuurr** [reply]
> You can prespecify a duration under Options > General > Temporary Override. If you leave it blank, then you get the prompt to enter a duration at the time the override is activated.

Ok fine! Sorry i didn't get this right when I was searching before posting.  Perhaps would it be possible to make it more configurable and granular as I have described my idea?

**Reply by proginosko** [reply]
It would be easy to add a drop-down selection to the text box in the override page, as with the lockdown page. It wouldn't be much more work to add an option to set a maximum duration for the override.

As for repeated overrides, note that there is already an option to limit the number of overrides per hour/day/week.

**Reply by slepto** [reply]
The drop-down selection idea could be very helpful if it's your trusted friend initiating the override for you.  The true addict needs to not be given any amount of leeway.  It would be great if there could be more than one way to initiate an override -- a pin set up for the trusted friend (can be more granular level of control over how long the override should last) vs. the long string of random characters with only a single relatively short duration override for the user him/herself.  The user doesn't know the pin, but can negotiate with the friend for a longer override.
